ui/pages/confirmations/components/simulation-details/formatAmount.ts

```diff
@@ -4,11 +4,51 @@ import {
   DEFAULT_PRECISION,
 } from '../../../../hooks/useCurrencyDisplay';
 
+const MAX_ELLIPSIS_LEFT_DIGITS = 15;
+
 // The number of significant decimals places to show for amounts less than 1.
 const MAX_SIGNIFICANT_DECIMAL_PLACES = 3;
 
 const ZERO_DISPLAY = '0';
 
+/**
+ * This function receives an formatted number and will append an ellipsis if the number of digits
+ * is greater than MAX_LEFT_DIGITS. Currently, we're only supporting en-US format. When we support
+ * i18n numbers, we'll need to update this method to support i18n.
+ *
+ * This method has been designed to receive results of formatAmount.
+ *
+ * There is no need to format the decimal portion because formatAmount shaves the portions off
+ * accordingly.
+ *
+ * @param amountText
+ * @param maxLeftDigits
+ */
+export function ellipsisAmountText(
+  amountText: string,
+  maxLeftDigits: number = MAX_ELLIPSIS_LEFT_DIGITS,
+): string {
+  const [integerPart] = amountText.split('.');
+  const cleanIntegerPart = integerPart.replace(/,/gu, '');
+
+  if (cleanIntegerPart.length > maxLeftDigits) {
```

Comment:
Could we re-use Goktug's `shortenString` function here?

From `ui/helpers/utils/util.js`.

Reply:
I think it’s better to keep it as is since this requires more logic to count the digits and preserve the “,”

Reply:
Sorry, I lost the design and issue ticket for this and ended up created a new ticket. Now that I've seen the relevant issue, we needed to do max 15 characters and not digits
https://github.com/MetaMask/MetaMask-planning/issues/2845

Updated to use `shortenString` here: https://github.com/MetaMask/metamask-extension/pull/26458